src/Components/Slider/Slider.js

```diff
@@ -0,0 +1,158 @@
+import React, { useState, useEffect, useRef } from 'react';
+import styled from 'styled-components';
+import { SliderContainer } from './SliderContainer';
+import { Slide } from './Slide';
+import { Arrow } from './Arrow';
+import { Dots } from './Dots';
+
+const SliderCSS = styled.div`
+  position: relative;
+  height: 100vh;
+  width: 100vw;
+  margin: 0 auto;
+  overflow: hidden;
+  white-space: nowrap;
+`
+
+const getWidth = () => window.innerWidth
+
+/**
+ * @function Slider
+ */
+const Slider = props => {
+
+  const slides  = Object.values(props);
+
+  const firstSlide = slides[0];
+  const secondSlide = slides[1];
+  const lastSlide = slides[slides.length - 1];
+
+  const [state, setState] = useState({
+    activeSlide: 0,
+    translate: getWidth(),
+    transition: 0.45,
+    transitioning: false,
+    _slides: [lastSlide, firstSlide, secondSlide]
+  });
+
+  const { activeSlide, translate, _slides, transition, transitioning } = state;
+
+  const autoPlayRef = useRef();
+  const transitionRef = useRef();
+  const resizeRef = useRef();
+  const sliderRef = useRef();
+  const throttleRef = useRef();
+
+
+  useEffect(() => {
+    autoPlayRef.current = nextSlide
+    transitionRef.current = smoothTransition
+    resizeRef.current = handleResize
+    throttleRef.current = throttleArrows
+  })
+
+  useEffect(() => {
+    const slider = sliderRef.current
+
+    const smooth = e => {
+      if (e.target.className.includes('SliderContent')) {
+        transitionRef.current()
+      }
+    }
+
+    const resize = () => {
+      resizeRef.current()
+    }
+
+    const throttle = e => {
+      if (e.target.className.includes('SliderContent')) {
+        throttleRef.current()
+      }
+    }
+
+    const transitionStart = slider.addEventListener('transitionstart', throttle)
+    const transitionEnd = slider.addEventListener('transitionend', smooth)
+    const onResize = window.addEventListener('resize', resize)
+
+    return () => {
+      slider.removeEventListener('transitionend', transitionStart)
+      slider.removeEventListener('transitionend', transitionEnd)
+      window.removeEventListener('resize', onResize)
```

Comment:
This won't remove the listeners, as the `removeEventListener` method requires the actual function that was used to register the listener in the first place. Notice that `addEventListener` doesn't return.

```javascript
function myFun() {
   // does something
}

dom.addEventListener('event', myFun);

dom.removeEventListener('event', myFun);
```